etna/models/nn/rnn.py

```diff
@@ -0,0 +1,223 @@
+        hidden_size:
+            size of the hidden state
+        lr:
+            learning rate
+        loss:
+            loss function
+        train_batch_size:
+            batch size for training
+        test_batch_size:
+            batch size for testing
+        trainer_params:
+            Pytorch ligthning  trainer parameters (api reference :py:class:`pytorch_lightning.trainer.trainer.Trainer`)
+        train_dataloader_params:
+            parameters for train dataloader like sampler for example (api reference :py:class:`torch.utils.data.DataLoader`)
+        test_dataloader_params:
+            parameters for test dataloader
+        val_dataloader_params:
+            parameters for validation dataloader
+        split_params:
+            parameters for torch dataset split for train-test splitting ``{"train_size": `float`, "generator": `Optional[torch.Generator]`}``
+        optimizer_params:
+            parameters for optimizer for Adam optimizer (api reference :py:class:`torch.optim.Adam`)
+        """
+        super().__init__(
+            encoder_length=encoder_length,
+            decoder_length=decoder_length,
+            train_batch_size=train_batch_size,
+            test_batch_size=test_batch_size,
+            train_dataloader_params={} if train_dataloader_params is None else train_dataloader_params,
+            test_dataloader_params={} if test_dataloader_params is None else test_dataloader_params,
+            val_dataloader_params={} if val_dataloader_params is None else val_dataloader_params,
+            trainer_params={} if trainer_params is None else trainer_params,
+            split_params={} if split_params is None else split_params,
+        )
+        self.num_layers = num_layers
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.loss = torch.nn.MSELoss() if loss is None else loss
+        self.layer = nn.LSTM(
+            num_layers=self.num_layers, hidden_size=self.hidden_size, input_size=self.input_size, batch_first=True
+        )
+        self.projection = nn.Linear(in_features=self.hidden_size, out_features=1)
+        self.lr = lr
+        self.optimizer_params = {} if optimizer_params is None else optimizer_params
+
+    def forward(self, x: Batch, *args, **kwargs):  # type: ignore
+        """Forward pass.
+
+        Parameters
+        ----------
+        x:
+            batch of data
+
+        Returns
+        -------
+        :
+            forecast with shape (batch_size, decoder_length, 1)
+        """
+        encoder_real = x["encoder_real"].float()  # (batch_size, encoder_length-1, input_size)
+        decoder_real = x["decoder_real"].float()  # (batch_size, decoder_length, input_size)
+        decoder_length = decoder_real.shape[1]
+        output, (h_n, c_n) = self.layer(encoder_real)
+        forecast = torch.zeros(
+            size=(decoder_real.shape[0], decoder_real.shape[1], 1), dtype=decoder_real.dtype, device=decoder_real.device
```

Comment:
May be add create the variable batch_size and use batch_size, decoder_lenght here

Reply:
Or actually use `torch.zeros_like(batch["decoder_target"])`